test/unit/plugin/v2/installer_test.rb

```diff
@@ -0,0 +1,414 @@
+    # @ruby_abi_version = (RUBY_VERSION.split('.')[0,2] << '0').join('.')
+    @ruby_abi_version = (Gem.ruby_version.segments[0, 2] << 0).join('.')
+
+    @installer = Inspec::Plugin::V2::Installer.instance
+    reset_globals
+    WebMock.disable_net_connect!(allow: 'api.rubygems.org')
+  end
+
+  def teardown
+    reset_globals
+
+    # We use the 'empty' config dir for exercising a lot of installs.
+    # Purge it after every test.
+    unless ENV['INSPEC_TEST_PRESERVE_PLUGIN']
+      Dir.glob(File.join(@config_dir_path, 'empty', '*')).each do |path|
+        next if path.end_with? '.gitkeep'
+        FileUtils.rm_rf(path)
+      end
+    end
+
+    # TODO: may need to edit the $LOAD_PATH, if it turns out that we need to "deactivate" gems after installation
+  end
+end
+
+#-----------------------------------------------------------------------#
+# basics
+#-----------------------------------------------------------------------#
+class PluginInstallerBasicTests < MiniTest::Test
+  include InstallerTestHelpers
+
+  # it's a singleton
+  def test_it_should_be_a_singleton
+    klass = Inspec::Plugin::V2::Installer
+    assert_equal klass.instance, klass.instance, "Calling instance on the Installer should always return the same object"
+    assert_raises(NoMethodError, 'Installer should have a private constructor') { klass.new }
+  end
+
+  # it should know its gem path
+  def test_it_should_know_its_gem_path_with_a_default_location
+    ENV['HOME'] = File.join(@config_dir_path, 'fakehome')
+    expected = File.join(ENV['HOME'], '.inspec', 'gems', @ruby_abi_version)
+    assert_equal expected, @installer.gem_path
+  end
+
+  def test_it_should_know_its_gem_path_with_a_custom_config_dir_from_env
+    ENV['INSPEC_CONFIG_DIR'] = File.join(@config_dir_path, 'empty')
+    expected = File.join(ENV['INSPEC_CONFIG_DIR'], 'gems', @ruby_abi_version)
+    assert_equal expected, @installer.gem_path
+  end
+end
+
+#-----------------------------------------------------------------------#
+# Installing
+#-----------------------------------------------------------------------#
+class PluginInstallerInstallationTests < MiniTest::Test
+  include InstallerTestHelpers
+
+  def test_refuse_to_install_gems_with_wrong_prefix
+    ENV['INSPEC_CONFIG_DIR'] = File.join(@config_dir_path, 'empty')
+    assert_raises(Inspec::Plugin::V2::InstallError) { @installer.install('ordinal_array')}
+  end
+
+  def test_install_a_gem_from_local_file
+    ENV['INSPEC_CONFIG_DIR'] = File.join(@config_dir_path, 'empty')
+    gem_file = File.join(@plugin_fixture_pkg_path, 'inspec-test-fixture-0.1.0.gem')
+    @installer.install('inspec-test-fixture', gem_file: gem_file)
+
+    # Installing a gem places it under the config dir gem area
+    spec_path = File.join(@installer.gem_path, 'specifications', 'inspec-test-fixture-0.1.0.gemspec')
+    assert File.exist?(spec_path), 'After installation from a gem file, the gemspec should be installed to the gem path'
```

Comment:
Might want to say something about how gem explodes the archive and stores a copy of the specification in a different location.